doc/src/explanation/active-deprecations.md

```diff
@@ -76,10 +76,28 @@ SymPy deprecation warnings.
 
 ## Version 1.14
 
+(deprecated-tensorproduct-simp)=
+### Deprecated tensor_product_simp from physics.quantum
+
+The ``tensor_product_simp`` function in the ``sympy.physics.quantum``
+module has been deprecated along with two helper functions,
+``tensor_product_simp_Mul`` and ``tensor_product_simp_Pow``. The 
+transformations performed by these functions are now applied
+automatically to all quantum expressions in the new
+``sympy.physics.quantum.transforms`` module.
+
+If you are using these functions in your code, you can remove them as
+they are now reduntant.
```

Comment:
Is it likely that downstream library code uses these functions?

If so I think it is unfriendly to start emitting a deprecation warning immediately. The functions can continue to work even if they are not needed any more and then when the new functionality has been available in a few releases it is a better time to add the warning.

Reply:
The old implementations of these functions are both slower, duplicative, and not as correct/complete as the automatics transforms. I am a bit hesitant to change the functions to a pass through without telling users. At the same time, leaving the original implementation and not warning leaves them using a function that isn't needed and doesn't work as well as they might like. Thoughts on which way to go?

Reply:
Also, I don't know that these are heavily used in downstream library code as they don't fully work as one would want.